### PR TITLE
Implement methods for testing and unwrapping result

### PR DESCRIPTION
**Pull request by marshtompsxd**
Implement exec methods for testing and unwrapping result, similar to option.

### CODE REVIEW

File: source/pervasive/result.rs [hunk 1]
```diff
@@ -1,9 +1,79 @@
...
+    // A more-readable synonym for get_Ok_0().
+    #[verifier(inline)]
+    pub open spec fn spec_unwrap_ok(self) -> T
+    recommends self.is_Ok()
+    {
+        self.get_Ok_0()
+    }
+
+    #[verifier(when_used_as_spec(spec_unwrap_ok))]
+    pub fn unwrap_ok(self) -> (t: T)
```
Comment (utaal) [inline source/pervasive/result.rs:45]: Can you please rename this to `unwrap`, to be consistent with https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap ?

Comment (marshtompsxd) [reply source/pervasive/result.rs]: Updated

**Comment by utaal** [comment]
Can you please add a couple of tests of the intended usage of these, in particular of the `when_used_as_spec` functions? (Sorry I didn't catch it the first time.)

**Comment by marshtompsxd** [comment]
Yes. Should I add tests in `rust_verify_test`?

> in particular of the when_used_as_spec functions?

I added it because I see similar things in option.rs. If this is undesired I can also remove it.

**Review by utaal — APPROVED** [review]
Thank you! Please use "Squash and merge" to merge.

**Pull request merged**